pkg/build/api/helpers.go

```diff
@@ -75,3 +78,58 @@ func HasTriggerType(triggerType BuildTriggerType, bc *BuildConfig) bool {
 	matches := FindTriggerPolicy(triggerType, bc)
 	return len(matches) > 0
 }
+
+// RecordStageAndStepInfo records details about each build stage and step
+func RecordStageAndStepInfo(stages []StageInfo, stageName StageName, stepName StepName, startTime unversioned.Time, endTime unversioned.Time) []StageInfo {
+	var stepDuration int64
+
+	// Make sure that the stages slice is initialized
+	if len(stages) == 0 {
+		stages = make([]StageInfo, 0)
```

Comment:
looks like it works fine, though i have to admit i'm slightly surprised and discomfited myself:
https://play.golang.org/p/_It7Pr3Tp2